File: plugins/code-metrics/scripts/fixtures/sources/CmSample.cs
// Fixture source for the code-metrics suites: a C# file, so the deferred
// dotnet lane has something to detect and report as deferred.
namespace Fixture;

public static class Sample
{
    public static string Classify(int value)
    {
        if (value > 0)
        {
            return "positive";
        }
        return value == 0 ? "zero" : "negative";
    }
}

File: plugins/code-metrics/scripts/fixtures/sources/cm_sample.py
"""Fixture source for the code-metrics suites.

One function with two branches and a nested helper, a comment, and blank
lines, so every collector has something to report. Never imported by the
plugin; kept lint-clean on purpose.
"""


def classify(value: int) -> str:
    """Return a coarse label for value."""

    def inner(flag: bool) -> str:
        # a nested function, so the range join has a nested range to subtract
        return "positive" if flag else "non-positive"

    if value > 0:
        return inner(True)
    if value == 0:
        return "zero"
    return inner(False)
